### ISSUE TITLE
Make it possible to override the synchronisation ID

### ISSUE THREAD

**Issue by kjelljorner** [pradyunsg/sphinx-inline-tabs#18]
Synchronization is obviously great for many purposes, but sometimes this is not something that you want. Take this example where I want to give input and (long) output in separate tabs. At the moment, the only way to prevent all tabs from switching is to rename the tab (Input 1, Input 2 etc.), which also changes its caption. I think a better way to do this would be to separate the caption from the name/label.
![image](https://user-images.githubusercontent.com/36157530/115252654-932c1880-a123-11eb-9d6c-cad341ba08f7.png)

**Comment by aantn** [comment]
I also have a use-case which requires disabling synchronization. Its a very nice feature but sometimes the complete opposite of what you want.

**Comment by astrsh** [comment]
Agreed, I'm having issues with not being able to turn off synchronization where it is not necessary, as changing tabs which are meant to be swapped back and forth causes the entire page to shift up/down when many synchronized tabs exist.

**Comment by aantn** [comment]
I switched to sphinx-design because of this. I still love this extension
though 😍

**Comment by kalekundert** [comment]
In case this is useful to anyone else, I completely turned off synchronization for my project by adding the following to `conf.py`:
```python
from pathlib import Path

def remove_tabs_js(app, exc):
    if app.builder.format == 'html' and not exc:
        tabs_js = Path(app.builder.outdir) / '_static' / 'tabs.js'
        tabs_js.unlink()

def setup(app):
    app.connect('build-finished', remove_tabs_js)
```

**Comment by pradyunsg** [comment]
FWIW, I'm on board for this idea and would appreciate a PR to add this!

The design I have in mind is:

- A `:sync: str` on `tab` directive, allowing authors to override behaviours. Error out on multiple tabs having the same sync key within a single tabset (ideally, with file + line number).
- A `:no-sync:` on `.. tab` directive, to opt-out on specific tabs.
- `tabs_default_sync_behavior` configuration option to configure the default behaviour with the initial supported values being `"tab-title"` (default) and `"none"`. Error out on all other values.

**Comment by cdce8p** [comment]
I'd like to disable the sync for some pages in the mypy docs.

Opened a PR with an initial implementation as spec'd in https://github.com/pradyunsg/sphinx-inline-tabs/issues/18#issuecomment-1627685795. Just added one more config option to disable the sync for specific labels globally.
```py
tabs_no_sync_labels = {"One", "Two"}
```

_To unblock CI, I opened #49._
